Code/Tools/AssetProcessor/native/ui/BuilderData.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#include <native/ui/BuilderData.h>
+#include <utilities/AssetUtilEBusHelper.h>
+#include <native/ui/BuilderDataItem.h>
+#include <AzToolsFramework/AssetDatabase/AssetDatabaseConnection.h>
+#include <AzCore/StringFunc/StringFunc.h>
+
+namespace AssetProcessor
+{
+    BuilderData::BuilderData(AZStd::shared_ptr<AzToolsFramework::AssetDatabase::AssetDatabaseConnection> dbConnection, QObject* parent)
+        : m_dbConnection(dbConnection)
+        , QObject(parent)
+    {
+    }
+
+    //! This method runs when this model is initialized. It gets the list of builders, gets existing stats about analysis jobs and
+    //! processing jobs, and matches stats with builders and save them appropriately for future use.
+    void BuilderData::Reset()
```

Comment:
maybe these comments should go into the BuilderData.h file?

Reply:
That makes sense. I will move them to the header file.

Reply:
resolved